### PR TITLE
fix: Casting issue due to MariaDB not included in inheritance schema anymore

### LINKED ISSUE #2887: TranslationWalker CAST fails for a string fieldMapping type on MariaDB
With DBAL 4.x, an issue that was fixed a long time ago (#1011, #1024) has arisen again, specifically for MariaDB.

In the original fix (#1025), no cast is done if the current platform is an instance of MySqlPlatform, which MariaDBPlatform extended.

Unfortunately:

- In DBAL 3.x: `class MariaDBPlatform extends MySQLPlatform`
- In DBAL 4.x: `class MariaDBPlatform extends AbstractMySQLPlatform`

In effect, the check doesn't work anymore for MariaDB.

### PR DESCRIPTION
**Pull request by Pixelshaped**
Hi there!

Fixes #2887

### CODE REVIEW

**Comment by Pixelshaped** [comment]
Converting to draft as I have to make sure this still works with DBAL3

**Review by mbabker — APPROVED** [review]

**Comment by phansys** [comment]
Thanks @Pixelshaped.

Could you please add the changelog entry for the fix?

**Comment by Pixelshaped** [comment]
Not sure I did well but the entry is added!

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ a release.
 ---
 
 ## [Unreleased]
+### Fixed
+- Fix regression with `doctrine/dbal` >= 4.0 that caused MariaDB to improperly try to CAST to LONGTEXT in TranslationWalker (issue #2887)
```
Comment (phansys) [inline CHANGELOG.md:22]:
```suggestion
- Fix regression with `doctrine/dbal` >= 4.0 that caused MariaDB to improperly attempt LONGTEXT casting in `TranslationWalker` (issue #2887)
```

**Review by phansys — APPROVED** [review]

**Comment by Pixelshaped** [comment]
Is there any ETA for release?

**Pull request merged**